test/format/replay.c

```diff
@@ -0,0 +1,543 @@
+     * in certain ways (e.g. operations with 10000 timestamps of a truncate would be
+     * forced to stay out of its table), there still would be a lot of details, and
+     * some rethink of our lane strategy. Even getting this to work, we would have
+     * a truncate that had the whole table to itself, which doesn't seem like an
+     * effective test.
+     */
+    if (op == TRUNCATE)
+        return (false);
+
+    /*
+     * We don't permit modify operations with predictable replay.
+     *
+     * The problem is read timestamps. As currently implemented, the read timestamp selected is
+     * variable, based on the state of other threads and their progress with other timestamped
+     * operations. And if two changes are made to the same key in a short amount of time, if the
+     * second operation were to be performed sometimes with a read timestamp before the first
+     * operation, and sometimes with a read timestamp after the first operation, then the results
+     * would be variable.
+     *
+     * We could track recent operations on a key (in its lane, for instance), but when we realize
+     * the read timestamp isn't recent enough, we would need to wait for the stable timestamp to
+     * move forward (and our waiting can affect/delay other thread's operations as well). Having the
+     * stable timestamp move forward is the only way our read timestamp can progress.
+     *
+     * Another possibility that also involves tracking recent operations on a key would be to
+     * disallow modifies that occur within, say 10000 timestamps of a previous write operation on
+     * the same key. Those modifies could be silently converted to reads, for instance. If our read
+     * timestamp was greater than 10000 timestamps behind, we'd still need to wait for the stable
+     * timestamp to catch up.
+     */
+    if (op == MODIFY)
+        return (false);
+
+    /*
+     * FIXME-WT-10570. We don't permit remove operations with predictable replay.
+     *
+     * This should be something we can and should fix. The problem may be similar to the problem
+     * with modify, where having a varying read timestamp can cause different results for different
+     * runs.
+     */
+    if (op == REMOVE)
+        return (false);
+
+    return (true);
+}
+
+/*
+ * replay_pick_timestamp --
+ *     Pick the next timestamp for this operation. That timestamp is used for any commits and also
+ *     determines which 'lane' we are in, to prevent races from occurring on operations on a single
+ *     key. Also, by using the timestamp to seed the random number generators, it also determines
+ *     precisely the nature of the operation.
+ */
+static void
+replay_pick_timestamp(TINFO *tinfo)
+{
+    uint64_t replay_seed, stop_ts, ts;
+    uint32_t lane;
+    bool in_use;
+
+    /*
+     * Choose a unique timestamp for commits. When we do predictable replay. If we're here with the
+     * 'again' flag, we already have a timestamp picked for us.
```

Comment:
To get rid of the quotes how about rewording like:
If the field for replaying again is set, we already have ...

Reply:
Fixed.